lib/sidekiq_flow/adapters/legacy_storage.rb

```diff
@@ -1,126 +1,133 @@
-            connection_pool.with do |redis|
-              redis.hmset(
-                workflow_key,
-                [:klass, workflow.klass, :attrs, workflow.to_json] + workflow.tasks.map { |t| [t.klass, t.to_json] }.flatten
-              )
-            end
-          end
-      
-          def store_task(task)
-            connection_pool.with do |redis|
-              redis.hset(find_workflow_key(task.workflow_id), task.klass, task.to_json)
-            end
-            return unless workflow_succeeded?(task.workflow_id)
-            succeed_workflow(task.workflow_id)
-          end
-      
-          def find_workflow(workflow_id)
-            connection_pool.with do |redis|
-              workflow_redis_hash = redis.hgetall(find_workflow_key(workflow_id))
-              raise WorkflowNotFound if workflow_redis_hash.empty?
-              Workflow.from_redis_hash(workflow_redis_hash)
-            end
-          end
-      
-          def destroy_workflow(workflow_id)
-            connection_pool.with do |redis|
-              redis.del(find_workflow_key(workflow_id))
-            end
-          end
-      
-          def destroy_succeeded_workflows
-            workflow_keys = find_workflow_keys(succeeded_workflow_key_pattern)
-            return if workflow_keys.empty?
-            connection_pool.with do |redis|
-              redis.del(*workflow_keys)            
-            end
-          end
-      
-          def find_workflow_keys(pattern=workflow_key_pattern)
-            connection_pool.with do |redis|
-              redis.scan_each(match: pattern, count: SCAN_COUNT).to_a
-            end
-          end
+  module Adapters
+    class LegacyStorage
+      SCAN_COUNT = 2000
 
-          def find_workflow_key(workflow_id)
-            key_pattern = "#{configuration.namespace}.#{workflow_id}_*"
-    
-            find_first(key_pattern)
-          end
-      
-          def connection_pool
-            @connection_pool ||= ConnectionPool.new(size: configuration.concurrency, timeout: configuration.timeout) do
-              Redis.new(url: configuration.redis_url)
-            end
-          end
+      attr_reader :configuration, :connection_pool
 
-          def find_first(key_pattern)
-            result = nil
-      
-            connection_pool.with do |redis|
-              cursor = "0"
-              loop do
-                cursor, keys = redis.scan(
-                          cursor,
-                          match: key_pattern,
-                          count: SCAN_COUNT
-                        )
-      
-                result = keys.first
-      
-                break if (result || cursor == "0")
-              end
-            end
-      
-            result
-          end
-      
-          private
+      def initialize(configuration)
+        @configuration = configuration
+        @connection_pool ||= ConnectionPool.new(size: configuration.concurrency, timeout: configuration.timeout) do
```

Comment:
Just move back the connection pool to its original method definition for now as that change will be on a separate ticket.